packages/datadog-plugin-kafkajs/src/producer.js

```diff
@@ -1,12 +1,51 @@
 'use strict'
 
 const ProducerPlugin = require('../../dd-trace/src/plugins/producer')
+const { getPathwayHash, encodePathwayContext, decodePathwayContext } = require('./hash')
 
 class KafkajsProducerPlugin extends ProducerPlugin {
   static get id () { return 'kafkajs' }
   static get operation () { return 'produce' }
 
   start ({ topic, messages }) {
+    const env = this.tracer._env
+    const service = this.tracer._service
+
+    let pathwayCtx
+    if (this.config.dsmEnabled !== 'disabled') {
+      const active = this.activeSpan
+      let parentHash
+      let originTimestamp
+      let prevTimestamp
+      const currentTimestamp = new Date().now() * 1000 // nanoseconds
+      const checkpointString = getCheckpointString(service, env, topic)
+      if (active) {
+        const context = active.context()
+        const rootSpan = context._trace.started[0];
+        [parentHash, originTimestamp, prevTimestamp] = decodePathwayContext(rootSpan._spanContext._tags.pathwayHash)
+      } else {
+        parentHash = Buffer.from('0000000000000000', 'hex')
```

Comment:
Since we don't mutate these buffers we could cache this value in the outer scope. Basically add this to the top:

```js
const ENTRY_PARENT_HASH = Buffer.from('0000000000000000', 'hex')
// and later ...
parentHash = ENTRY_PARENT_HASH
```